specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/examples/CreateOrUpdateASimpleGalleryImageVersionWithVHD.json

```diff
@@ -0,0 +1,215 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "myResourceGroup",
+    "api-version": "2020-09-30",
+    "galleryName": "myGalleryName",
+    "galleryImageName": "myGalleryImageName",
+    "galleryImageVersionName": "1.0.0",
+    "galleryImageVersion": {
+      "location": "West US",
+      "properties": {
+        "publishingProfile": {
+          "targetRegions": [
+            {
+              "name": "West US",
+              "regionalReplicaCount": 1,
+              "encryption": {
+                "osDiskImage": {
+                  "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myDiskEncryptionSet"
+                },
+                "dataDiskImages": [
+                  {
+                    "diskEncryptionSetId": "/subscriptions/{subscriptionId}/resourceGroups/myResourceGroup/providers/Microsoft.Compute/diskEncryptionSet/myOtherDiskEncryptionSet",
+                    "lun": 1
+                  }
+                ]
+              }
+            },
+            {
+              "name": "East US",
+              "regionalReplicaCount": 2,
+              "storageAccountType": "Standard_ZRS"
+            }
+          ]
+        },
+        "storageProfile": {
+          "source": {
+            "id": "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.Compute/vhd/{vhdName}",
```

Comment:
Is '/Microsoft.Compute/vhd/' a valid resource? Which team owns it? I thought only the uri was a valid input for blob/VHD input

Reply:
this should actually be the storage account resource ID

Reply:
ex. /subscriptions/97f78232-382b-46a7-8a72-964d692c4f3f/resourceGroups/image-copy-rg/providers/Microsoft.Storage/storageAccounts/brazilsoutheastze5qof69j

Reply:
Gotcha. so for a blob input, does both the resourceId (of the storage account) and the sas uri have to be provided?

Reply:
close, both the storage account resource ID and the vhd uri will have to be provided. CAPS will have to validate the storage accounts match between those two

Reply:
just curious, why is the validation of the storage accounts necessary?

Reply:
only the ID field has linked access checks. So by having storage account there, we can leverage ARM to verify caller has access. it is not possible to do linked access checks on VHDs. So the combination of the fields ensure we can have both access checking + accept vhd as a source